### ISSUE TITLE
Automatically add `-- ` when pressing enter in a comment

### ISSUE THREAD

**Issue by Sir4ur0n** [haskell/haskell-language-server#336]
A minor but nice feature I would like to propose is to automatically add `-- ` on the new line when pressing "enter" in a comment/documentation.

Example (cursor position represented by `[]`) below.

# Current behavior
```haskell
-- | Somme doc[]
foo = "hello world"
```
Press enter
```haskell
-- | Somme doc
[]
foo = "hello world"
```

# Proposed behavior
```haskell
-- | Somme doc[]
foo = "hello world"
```
Press enter
```haskell
-- | Somme doc
-- []
foo = "hello world"
```

I think this default behavior makes much more sense for developers.

To be discussed: should we do the same when in a regular comment? I'd say "no" but I am not strongly opinionated on this one.

Cheers

**Comment by bubba** [comment]
I think this is going to be specific to the client/editor, as far as I'm aware LSP doesn't have anything in the specification for auto-indenting. It has auto-formatting but that seems to be something slightly different. If this is in VS Code then maybe https://github.com/JustusAdam/language-haskell might be able to handle this

**Comment by Sir4ur0n** [comment]
Fair point @bubba, it slipped my mind that it probably doesn't belong in the HLS backend, thank you!

That being said, any reason why this wouldn't belong in https://github.com/haskell/vscode-haskell ?

**Comment by bubba** [comment]
vscode-haskell itself doesn't handle anything outside of setting up + spinning up the language server, i.e. the commenting and syntax highlighting is done by that https://github.com/JustusAdam/language-haskell dependency. Although perhaps its possible for vscode-haskell to specifically handle the commenting stuff?
